protocol/xmrmaker/claim.go

```diff
@@ -135,41 +193,93 @@ func claimRelayer(
 		return ethcommon.Hash{}, err
 	}
 
-	abi, err := abi.JSON(strings.NewReader(contracts.SwapFactoryMetaData.ABI))
+	calldata, err := getClaimTxCalldata(relayerCommission, contractSwap, secret)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	feeValue, err := calculateRelayerCommission(contractSwap.Value, relayerCommission)
+	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	calldata, err := abi.Pack("claimRelayer", *contractSwap, secret, feeValue)
+	err = waitAndCheck(ctx, ec, txHash)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
+	return txHash, nil
+}
+
+func waitAndCheck(ctx context.Context, ec *ethclient.Client, txHash ethcommon.Hash) error {
```

Comment:
This is a large package making the scope of this function large.  What do you think of the name `waitForClaimReceipt` or something similar?

Reply:
How do we know that the transaction hash that the relayer gave us was for our claim?

Reply:
I can parse the tx calldata field to check it's ours, will add

Reply:
@dimalinux updated so that the `Claimed` log is checked

Reply:
it's not really useful to parse the calldata, there's too many fields and the tx might revert anyways, better to check the logs emitted from the tx